snuba/state/cache/redis/backend.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import uuid
+from typing import Callable, Optional, TypeVar
+
+from pkg_resources import resource_string
+from redis.exceptions import ResponseError
+
+from snuba.redis import RedisClientType
+from snuba.state import get_config
+from snuba.state.cache.abstract import Cache, ExecutionError, ExecutionTimeoutError
+from snuba.utils.codecs import Codec
+
+
+T = TypeVar("T")
+
+
+logger = logging.getLogger(__name__)
+
+
+RESULT_VALUE = 0
+RESULT_EXECUTE = 1
+RESULT_WAIT = 2
+
+
+class RedisCache(Cache[T]):
+    def __init__(
+        self, client: RedisClientType, prefix: str, codec: Codec[bytes, T]
```

Comment:
This also changed from `Codec[str, T]` to `Codec[bytes, T]`, apparently this was always wrong (per typeshed and also what I am seeing interacting with this locally.)